File: leetcode806.py
class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:
      result: List[int] = []
      ans: List[int] = []
      i: int = 0
      p: int = 0
      while i < len(s):
        if p + widths[ord(s[i]) - ord('a')] <= 100:
          p += widths[ord(s[i]) - ord('a')]
        else:
          result.append(p)
          p = widths[ord(s[i]) - ord('a')]
        i += 1
      result.append(p)
      return [len(result),result[len(result) - 1]]
      

File: leetcode806.cpp
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string s) {
      vector<int>ans;
      vector<int>result;
      int i = 0,p = 0;
      while(i < s.length()){
        if(p + widths[s[i] - 'a'] <= 100){
          p += widths[s[i] - 'a'];
        }else{
          result.push_back(p);
          p = widths[s[i] - 'a'];
        }
        i ++;
      }
      result.push_back(p);
      ans.push_back(result.size());
      ans.push_back(result[ans[0] - 1]);
      return ans;
    }
};
